integration/hive/src/main/java/org/apache/carbondata/hive/util/HiveCarbonUtil.java

```diff
@@ -0,0 +1,332 @@
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.metastore.HiveMetaHook;
+import org.apache.hadoop.hive.metastore.api.FieldSchema;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.log4j.Logger;
+
+public class HiveCarbonUtil {
+
+  private static final Logger LOGGER =
+      LogServiceFactory.getLogService(HiveCarbonUtil.class.getName());
+
+  public static CarbonLoadModel getCarbonLoadModel(Configuration tableProperties) {
+    String[] tableUniqueName = tableProperties.get("name").split("\\.");
+    String databaseName = tableUniqueName[0];
+    String tableName = tableUniqueName[1];
+    String tablePath = tableProperties.get("location");
+    String columns = tableProperties.get("columns");
+    String sortColumns = tableProperties.get("sort_columns");
+    String columnTypes = tableProperties.get("columns.types");
+    String partitionColumns = tableProperties.get("partition_columns");
+    String partitionColumnTypes = tableProperties.get("partition_columns.types");
+    if (partitionColumns != null) {
+      columns = columns + "," + partitionColumns;
+      columnTypes = columnTypes + ":" + partitionColumnTypes;
+    }
+    String[] columnTypeArray = splitSchemaStringToArray(columnTypes);
+    String complexDelim = tableProperties.get("complex_delimiter", "");
+    CarbonLoadModel carbonLoadModel =
+        getCarbonLoadModel(tableName, databaseName, tablePath, sortColumns, columns.split(","),
+            columnTypeArray, tableProperties);
+    carbonLoadModel.setCarbonTransactionalTable(true);
+    carbonLoadModel.getCarbonDataLoadSchema().getCarbonTable().setTransactionalTable(true);
+    for (String delim : complexDelim.split(",")) {
+      carbonLoadModel.setComplexDelimiter(delim);
+    }
+    return carbonLoadModel;
+  }
+
+  public static CarbonLoadModel getCarbonLoadModel(Properties tableProperties,
+      Configuration configuration) {
+    String[] tableUniqueName = tableProperties.getProperty("name").split("\\.");
+    String databaseName = tableUniqueName[0];
+    String tableName = tableUniqueName[1];
+    String tablePath = tableProperties.getProperty("location");
+    String columns = tableProperties.getProperty("columns");
+    String sortColumns = tableProperties.getProperty("sort_columns");
+    String[] columnTypes = splitSchemaStringToArray(tableProperties.getProperty("columns.types"));
+    String complexDelim = tableProperties.getProperty("complex_delimiter", "");
+    CarbonLoadModel carbonLoadModel =
+        getCarbonLoadModel(tableName, databaseName, tablePath, sortColumns, columns.split(","),
+            columnTypes, configuration);
+    for (String delim : complexDelim.split(",")) {
+      carbonLoadModel.setComplexDelimiter(delim);
+    }
+    return carbonLoadModel;
+  }
+
+  public static CarbonLoadModel getCarbonLoadModel(String tableName, String databaseName,
```

Comment:
can you use CarbonLoadModelBuilder to build it

Reply:
done